datalad/distribution/publish.py

```diff
@@ -256,49 +282,91 @@ def __call__(path=None, dataset=None, to=None, recursive=False, since=None,
-            # Note: This is about a gitpython issue as well as something about
-            # annex -> might mean, that we need to do it in case we pushed an annex branch only.
-            # Apparently, we can annex copy new files only, after this fetch. Figure it out!
-            ds.repo.fetch(remote=dest_resolved)
+            # upstream branch needed for update (merge) and subsequent push,
+            # in case there is no.
+            # no tracking branch yet?
+            set_upstream = track_branch is None
+
+            # publishing of `dest_resolved` might depend on publishing other
+            # remote(s) first:
+            # define config var name for potential publication dependencies
+            depvar = 'remote.{}.datalad-publish-depends'.format(dest_resolved)
+            for d in ds.config.get(depvar, []):
+                lgr.info("Dependency detected: '%s'" % d)
+                # Note: Additional info on publishing the dep. comes from within
+                # `ds.publish`.
+                ds.publish(path=path,
+                           to=d,
+                           since=since,
+                           skip_failing=skip_failing,
+                           recursive=recursive,
+                           recursion_limit=recursion_limit,
+                           git_opts=git_opts,
+                           annex_opts=annex_opts,
+                           annex_copy_opts=annex_copy_opts)
+
+            lgr.info("Publishing {0} to {1}".format(ds, dest_resolved))
+
+            # we now know where to push to:
+            # TODO: what to push? default: git push --mirror if nothing configured?
+            # consider also: --follow-tags, --tags, --atomic
+
+            # Note: git's push.default is 'matching', which possibly doesn't
+            # work for first
+            # time publication (a branch, that doesn't exist on remote yet)
+            # But if we want to respect remote.*.push entries, etc. we need to
+            # not pass a specific refspec (like active branch) to `git push`
+            # by default.
+
+            _log_push_info(ds.repo.push(remote=dest_resolved,
+                                        refspec=ds.repo.get_active_branch(),
+                                        set_upstream=set_upstream))
 
             published.append(ds)
 
             if publish_files or annex_copy_opts:
                 if not isinstance(ds.repo, AnnexRepo):
-                    raise RuntimeError(
-                        "Cannot publish content of something, that is not "
-                        "part of an annex. ({0})".format(ds))
+                    # incomplete, since `git push` was done already:
+                    raise IncompleteResultsError(
+                        (published, skipped),
+                        failed=publish_files,
+                        msg="Cannot publish content of something, that is not "
+                            "an annex. ({0})".format(ds))
```

Comment:
Why you can't? It is just that you don't have to do anything special to make it happen, bit you can publish them.
My main concern is this exception while publishing recursively set of datasets since if which are git. IMHO this exception will be a big obstacle and Skype be replaced with a simple warning at most